src/results_analyze/rules_analyze.py

```diff
@@ -0,0 +1,299 @@
+
+        appended_data = []
+
+        for (rule_text, rule_yml) in zip(files_rule, files_yml):
+            f_yml = open(rule_yml, 'r')
+            f_rule_text = open(rule_text, 'r')
+
+            df = pd.json_normalize(yaml.safe_load(f_yml))
+            rule_text_string = f_rule_text.read()
+            no_words = len(rule_text_string.split())
+            df["Rule_text"] = rule_text_string
+            df["num_words"] = no_words
+
+            appended_data.append(df)
+
+        appended_data = pd.concat(appended_data)
+
+        appended_data["rule_name"] = pd.Series(files_rule, index=appended_data.index)
+        appended_data.index = pd.RangeIndex(start=1, stop=appended_data.shape[0] + 1)
+
+        return appended_data
+
+    def load_lic_info_from_scancode(self):
+        """
+        From the scancode License Files, at `scancode/src/licensedcode/data/licenses/` load all License Text and .yml
+        files into a DataFrame.
+
+        :return appended_data: pd.DataFrames
+            A DataFrame containing all the Scancode License Information.
+        """
+
+        lic_files_all = []
+
+        for (dirpath, dirnames, filenames) in os.walk(self.lic_folder_path):
+            filenames.sort()
+            lic_files_all.extend(filenames)
+
+        files_notice = []
+        files_yml = []
+
+        for file in lic_files_all:
+            if file[-7:] == 'LICENSE':
+                files_notice.append(file)
+            elif file[-3:] == 'yml':
+                files_yml.append(file)
+
+        notice_set = set([file_name[:-8] for file_name in files_notice])
+        yml_set = set([file_name[:-4] for file_name in files_yml])
+
+        extra_yml_files = [name + '.yml' for name in list(yml_set - (notice_set & yml_set))]
+        files_yml_new = [e for e in files_yml if e not in extra_yml_files]
+
+        assert len(files_notice) == len(files_yml_new)
+
+        os.chdir(self.lic_folder_path)
+
+        appended_data_notice = []
+
+        for (notice_text, notice_yml) in zip(files_notice, files_yml_new):
+            f_yml = open(notice_yml, 'r')
+            f_notice_text = open(notice_text, 'r')
+
+            df = pd.json_normalize(yaml.safe_load(f_yml))
+            notice_text_string = f_notice_text.read()
+            no_words = len(notice_text_string.split())
+            df["Notice_text"] = notice_text_string
+            df["Notice_len"] = no_words
+
+            appended_data_notice.append(df)
+
+        appended_data_notice = pd.concat(appended_data_notice)
+
+        appended_data_notice["license_name"] = pd.Series(files_notice, index=appended_data_notice.index)
+        appended_data_notice.index = pd.RangeIndex(start=1, stop=appended_data_notice.shape[0] + 1)
+
+        return appended_data_notice
+
+    @staticmethod
+    def rule_compute_relevance(rule_df, threshold=THRESHOLD_COMPUTE_RELEVANCE):
+        """
+        Compute all the Relevance Values of Rules where it isn't given explicitly.
```

Comment:
Are relevance values and coverage values a scancode term?

Reply:
Yes. A similar function is there (acts on One Rule at a time) at -> `Rule.compute_relevance` in `scancode.licensedcode.models.py`. This acts on all the Rules at once.